src/probnum/filtsmooth/gaussfiltsmooth/kalman.py

```diff
@@ -69,19 +67,21 @@ def iterated_filtsmooth(
 
     def filtsmooth(
         self,
-        dataset: np.ndarray,
-        times: np.ndarray,
+        regression_problem: RegressionProblem,
         _previous_posterior: Optional[TimeSeriesPosterior] = None,
     ):
         """Apply Gaussian filtering and smoothing to a data set.
 
         Parameters
         ----------
-        dataset : array_like, shape (N, M)
-            Data set that is filtered.
-        times : array_like, shape (N,)
-            Temporal locations of the data points.
-            The zeroth element in times and dataset is the location of the initial random variable.
+        regression_problem: RegressionProblem
```

Comment:
you dont need the `: RegressionProblem` here, it should be filled in automatically based on the type hint